File: skills/vuln-discovery/evals/fixtures/python/ssrf_vulnerable.py
"""Vulnerable: SSRF via user-controlled URL in requests.get."""
import requests
from flask import Flask, request as req

app = Flask(__name__)

@app.route("/fetch")
def fetch_url():
    url = req.args.get("url")
    resp = requests.get(url)
    return resp.text

@app.route("/proxy")
def proxy():
    target = req.args.get("target")
    resp = requests.get(f"http://{target}/api/data")
    return resp.json()

File: skills/vuln-discovery/evals/fixtures/go/ssrf_vulnerable.go
// Vulnerable: SSRF via user-controlled URL passed to http.Get.
package main

import (
	"io"
	"net/http"
)

func fetchHandler(w http.ResponseWriter, r *http.Request) {
	url := r.URL.Query().Get("url")
	resp, err := http.Get(url)
	if err != nil {
		http.Error(w, err.Error(), 500)
		return
	}
	defer resp.Body.Close()
	io.Copy(w, resp.Body)
}
